crates/hir_def/src/nameres/collector.rs

```diff
@@ -1279,13 +1279,12 @@ impl DefCollector<'_> {
 
         for directive in &self.unresolved_imports {
             if let ImportSource::Import { id: import, use_tree } = &directive.import.source {
-                match (directive.import.path.segments().first(), &directive.import.path.kind) {
-                    (Some(krate), PathKind::Plain | PathKind::Abs) => {
-                        if diagnosed_extern_crates.contains(krate) {
-                            continue;
-                        }
+                if let (Some(krate), PathKind::Plain | PathKind::Abs) =
```

Comment:
I think this was intentional.

Reply:
how so? It doesn't really change the nesting level, though on another note match + if guard seems like the proper solution here

Reply:
But this isn't an `if let ... {} else {}`, just an if let

Reply:
Yeah, you're right.

Reply:
that was indeed intentional -- multiline conditions are not readable. In

```rust
if let (Some(krate), PathKind::Plain | PathKind::Abs) =
    (directive.import.path.segments().first(), &directive.import.path.kind)
```

you need to read a lot to understand what the heck are you even matching. With `match`, scrutinee comes first, which I personally find easier on the eyes: pattern in isolation doesn't make much sense, it only is meaningful with respect to a specific expression you pattern match. 

But no strong opinion here!